### PR TITLE
Logs (#158)

### LINKED ISSUE #158: Logs channel
Send warnings or other types of logs to a specific channel.

### PR DESCRIPTION
**Pull request by ostorc**
Fixes #158.

### CODE REVIEW

File: src/HonzaBotner.Discord/DiscordBot.cs [hunk 1]
```diff
@@ -18,17 +21,19 @@ internal class DiscordBot : IDiscordBot
...
 
         public DiscordBot(DiscordWrapper discordWrapper, EventHandler.EventHandler eventHandler,
-            CommandConfigurator configurator, IVoiceManager voiceManager)
+            CommandConfigurator configurator, IVoiceManager voiceManager, IOptions<DiscordConfig> discordOptions)
         {
             _discordWrapper = discordWrapper;
             _eventHandler = eventHandler;
             _configurator = configurator;
             _voiceManager = voiceManager;
+            _discordOptions = discordOptions;
```
Comment (tenhobi) [inline src/HonzaBotner.Discord/DiscordBot.cs]: The property should be `DiscordConfig`, not `IOptions<DiscordConfig>`, e.g. in the constructor it should be ` _discordOptions = discordOptions,Value;`.

File: src/HonzaBotner.Discord/DiscordBot.cs [hunk 2]
```diff
@@ -161,5 +176,19 @@ private Task Client_ChannelCreated(DiscordClient client, ChannelCreateEventArgs
...
+        private async Task ReportException(DiscordGuild guild, string source, Exception exception)
+        {
+            ulong logChannelId = _discordOptions.Value.LogChannelId;
+
+            if (logChannelId == default)
+            {
+                return;
+            }
+
+            DiscordChannel channel = guild.GetChannel(logChannelId);
```
Comment (tenhobi) [inline src/HonzaBotner.Discord/DiscordBot.cs:195]: `GetChannel` could fail, maybe trycatch it?

**Review by tenhobi — CHANGES_REQUESTED** [review]

File: src/HonzaBotner.Discord/DiscordBot.cs (hunk 2 above)
Comment (ostorc) [reply src/HonzaBotner.Discord/DiscordBot.cs:195]: This is error handling, if we fail here we are doomed anyway

Comment (jirkavrba) [reply src/HonzaBotner.Discord/DiscordBot.cs:195]: Hail satan elmo

Comment (tenhobi) [reply src/HonzaBotner.Discord/DiscordBot.cs:195]: Couldn't that also throw an exception, ... ? Like you would trigger another call of this function and again, again, ...😂

File: src/HonzaBotner/Controllers/AuthController.cs [hunk 3]
```diff
@@ -70,7 +70,7 @@ public async Task<ActionResult> Callback()
 
                 return Page(message, 200);
             }
-            catch (InvalidOperationException e)
+            catch (Exception e)
```
Comment (tenhobi) [inline src/HonzaBotner/Controllers/AuthController.cs:76]: I used generic `Exception` to catch all the errors. For example, running `https://localhost:5001/Auth/Callback?code=wtf` showed the error dump without this + `src/HonzaBotner.Services/CvutAuthorizationService.cs` changes.

**Review by tenhobi — DISMISSED** [review]

**Review by tenhobi — DISMISSED** [review]

**Review by tenhobi — APPROVED** [review]
LGTM

**Pull request merged**